src/nupic/algorithms/AnomalyLikelihood.hpp

```diff
@@ -0,0 +1,261 @@
+                        and 10,000 records
+  :param averagingWindow: integer number of records to average over
+  :param skipRecords: integer specifying number of records to skip when
+                      estimating distributions. If skip records are >=
+                      len(anomalyScores), a very broad distribution is returned
+                      that makes everything pretty likely.
+  :param verbosity: integer controlling extent of printouts for debugging
+
+                      0 = none
+                      1 = occasional information
+                      2 = print every record
+
+  :returns: vector of lilelihoods: , one for each aggregated point
+  **/
+    vector<Real> estimateAnomalyLikelihoods(vector<Real> anomalyScores, UInt skipRecords=0, UInt verbosity=0);
+
+  /**
+  Compute updated probabilities for anomalyScores using the given params.
+
+  :param anomalyScores: a list of records. Each record is a list with the
+
+  :param verbosity: integer controlling extent of printouts for debugging
+  :type verbosity: UInt
+
+  :returns: a vector of likelihoods, one for each aggregated point
+  **/
+    vector<Real>  updateAnomalyLikelihoods(vector<Real> anomalyScores, UInt verbosity=0);
+ /**
+  Given the normal distribution specified by the mean and standard deviation
+  in distributionParams (the distribution is an instance member of the class), 
+  return the probability of getting samples further from the mean. 
+  For values above the mean, this is the probability of getting
+  samples > x and for values below the mean, the probability of getting
+  samples < x. This is the Q-function: the tail probability of the normal distribution.
+  **/
+    Real32 tailProbability(Real32 x) const;
+  /**
+  :param sampleData:
+  :type sampleData: vector array //TODO of what? likelihoods? rawScores? ....?
+  :param performLowerBoundCheck:
+  :type performLowerBoundCheck: bool
+  :returns: A DistributionParams (struct) containing the parameters of a normal distribution based on
+      the ``sampleData``.
+  **/
+    DistributionParams estimateNormal(vector<Real> sampleData, bool performLowerBoundCheck=true);
+  /** returns parameters of a Null distribution **/
+    DistributionParams nullDistribution() const {
+    return DistributionParams("normal", 0.5, 1e6, 1e3);
+}
+
+    //private static methods
+static Real  computeLogLikelihood(Real likelihood)  {
+  /**
+    Compute a log scale representation of the likelihood value. Since the
+    likelihood computations return low probabilities that often go into four 9's
+    or five 9's, a log value is more useful for visualization, thresholding,
+    etc.
+   **/
+    // The log formula is:
+    //     Math.log(1.0000000001 - likelihood) / Math.log(1.0 - 0.9999999999)
+    return log(1.0000000001f - likelihood) / -23.02585084720009f;
+}
+
+static UInt calcSkipRecords_(UInt numIngested, UInt windowSize, UInt learningPeriod)  {
```

Comment:
This can be a function in the cpp file. It doesn't need to be a method in the header.